src/main/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocLexer.g4

```diff
@@ -37,8 +37,7 @@ import java.util.*;
 
 }
 
-LEADING_ASTERISK : ( (' '|'\t') {_tokenStartCharPositionInLine == 0}? ) (' '|'\t')* '*'
-      | '*' {_tokenStartCharPositionInLine == 0}?
+LEADING_ASTERISK : (' '|'\t')* '*'
```

Comment:
@rnveach Actaully it wasn't recognizing leading tokens after a newline so I had two options either to modify the entire rule or add the alternative `| '*' {previousTokenType = NEWLINE}?`

I chose the former and just tried to simplify the token def as much as I could based on the basic format of javadoc. Since the UTs seem to be insufficient, I was hoping to realize if there is any problem with this update, with the regression testing tool which is to be developed without which this PR isn't getting merged anyway.